### PR TITLE
provider/azurerm: Create azure App services

### PR DESCRIPTION
**Pull request by lstolyarov**
Example usage:
```
resource "azurerm_server_farm" "server_farm" {
  name = "${var.server_farm_name}"
  location = "${var.location}"
  resource_group_name = "${azurerm_resource_group.resource_group.name}"
  tier = "S3"
}

resource "azurerm_web_app" "web_app" {
  name = "${var.web_app_name}"
  resource_group_name = "${azurerm_resource_group.resource_group.name}"
  server_farm_id = "${azurerm_server_farm.server_farm.id}"
  location = "${var.location}"
  always_on = "true"
}
```

### CODE REVIEW

**Comment by lstolyarov** [comment]
@stack72 here is a pull request we discussed yesterday.

**Comment by worldspawn** [comment]
Are there any plans to support controlling application settings, web slots, instance counts, etc?

**Comment by rolandoldengarm** [comment]
Yep same here. We use app services a lot and I'm surprised it's not in there yet.

File: builtin/providers/azurerm/provider.go [hunk 1]
```diff
@@ -124,6 +124,9 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_sql_database":      resourceArmSqlDatabase(),
 			"azurerm_sql_firewall_rule": resourceArmSqlFirewallRule(),
 			"azurerm_sql_server":        resourceArmSqlServer(),
+
+			"azurerm_web_app":     resourceArmWebApp(),
+			"azurerm_server_farm": resourceArmServerFarm(),
```
Comment (lstolyarov) [inline builtin/providers/azurerm/provider.go]: Update the naming conventions. It should align with Microsoft product names now.

File: builtin/providers/azurerm/config.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/hashicorp/terraform/terraform"
 	riviera "github.com/jen20/riviera/azure"
+	"github.com/lstolyarov/azure-sdk-for-go/arm/web"
```
Comment (lstolyarov) [inline builtin/providers/azurerm/config.go]: Switched over to the official SDK as it is now working correctly.

File: builtin/providers/azurerm/resource_arm_web_app.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/lstolyarov/azure-sdk-for-go/arm/web"
+)
+
+func resourceArmWebApp() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmWebAppCreate,
+		Read:   resourceArmWebAppRead,
+		Update: resourceArmWebAppCreate,
```
Comment (lstolyarov) [inline builtin/providers/azurerm/resource_arm_web_app.go]: Updated for both App service and App service plan.

File: builtin/providers/azurerm/resource_arm_web_app.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+				Optional: true,
+				Default:  "",
+			},
+			"force_dns_registration": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"ttl_in_seconds": {
+				Type:     schema.TypeString,
```
Comment (lstolyarov) [inline builtin/providers/azurerm/resource_arm_web_app.go]: The SDK method expects a string, therefore, I left as a string in here.

**Comment by AtwooTM** [comment]
@tombuildsstuff , now that the Azure SDK has been fixed, any ETA on getting Terraform resource support for Azure App Services and App Service Plans soon?  Thanks in advance.

**Comment by tombuildsstuff** [comment]
Hi @lstolyarov / @AtwooTM 

Thanks for splitting this PR out into two :)

Just to give an update here - this PR is blocked on the following PR's:
 - #14004 - Upgrade to v9.1 of the Azure SDK for Go. This is needed to fix a serialisation bug in v8 of the SDK regarding App Services, and a regression in v9 of the SDK which means we're unable to upgrade.
 - #13634 - App Service Plans, which need to exist to create App Services :)

Once both of these are merged, we can proceed with this PR - apologies for the delay; and thanks for all the great work here :)

File: builtin/providers/azurerm/resource_arm_app_service.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			},
+			"force_dns_registration": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"ttl_in_seconds": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
```
Comment (lstolyarov) [inline builtin/providers/azurerm/resource_arm_app_service.go]: ttl_in_seconds is passed in as a string

https://godoc.org/github.com/Azure/azure-sdk-for-go/arm/web#AppsClient.CreateOrUpdate

**Comment by lstolyarov** [comment]
Migrated to https://github.com/terraform-providers/terraform-provider-azurerm/pull/2

**Pull request closed without merging**